### ISSUE TITLE
error when installing on Windows 11

### ISSUE THREAD

**Issue by gardc** [alliedmodders/ambuild#149]
```
PS C:\Users\dev\Downloads> pip install .\ambuild\
Processing c:\users\dev\downloads\ambuild
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [42 lines of output]
      Traceback (most recent call last):
        File "C:\Users\dev\AppData\Local\Programs\Python\Python312\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
          main()
        File "C:\Users\dev\AppData\Local\Programs\Python\Python312\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\dev\AppData\Local\Programs\Python\Python312\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\dev\AppData\Local\Temp\pip-build-env-0irrx5re\overlay\Lib\site-packages\setuptools\build_meta.py", line 355, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\dev\AppData\Local\Temp\pip-build-env-0irrx5re\overlay\Lib\site-packages\setuptools\build_meta.py", line 325, in _get_build_requires
          self.run_setup()
        File "C:\Users\dev\AppData\Local\Temp\pip-build-env-0irrx5re\overlay\Lib\site-packages\setuptools\build_meta.py", line 507, in run_setup
          super(_BuildMetaLegacyBackend, self).run_setup(setup_script=setup_script)
        File "C:\Users\dev\AppData\Local\Temp\pip-build-env-0irrx5re\overlay\Lib\site-packages\setuptools\build_meta.py", line 341, in run_setup
          exec(code, locals())
        File "<string>", line 26, in <module>
        File "C:\Users\dev\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\process.py", line 121, in start
          self._popen = self._Popen(self)
                        ^^^^^^^^^^^^^^^^^
        File "C:\Users\dev\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\context.py", line 224, in _Popen
          return _default_context.get_context().Process._Popen(process_obj)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\dev\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\context.py", line 337, in _Popen
          return Popen(process_obj)
                 ^^^^^^^^^^^^^^^^^^
        File "C:\Users\dev\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\popen_spawn_win32.py", line 94, in __init__
          reduction.dump(process_obj, to_child)
        File "C:\Users\dev\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\reduction.py", line 60, in dump
          ForkingPickler(file, protocol).dump(obj)
      _pickle.PicklingError: Can't pickle <function detect_distutils at 0x0000016E6E732F20>: attribute lookup detect_distutils on __main__ failed
      Traceback (most recent call last):
        File "<string>", line 1, in <module>
        File "C:\Users\dev\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\spawn.py", line 113, in spawn_main
          new_handle = reduction.duplicate(pipe_handle,
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\dev\AppData\Local\Programs\Python\Python312\Lib\multiprocessing\reduction.py", line 79, in duplicate
          return _winapi.DuplicateHandle(
                 ^^^^^^^^^^^^^^^^^^^^^^^^
      PermissionError: [WinError 5] Access is denied
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

Tried in user and admin powershell

**Comment by dvander** [comment]
This does look like a setuptools bug... Maybe try using command prompt or a different python/setuptools version.

**Comment by LifeDreamer24** [comment]
same here...

**Comment by LifeDreamer24** [comment]
Fixed, use py 1.10.0
https://www.python.org/downloads/release/python-3100/

**Comment by ghostt187** [comment]
try this 
pip install wheel setuptools pip --upgrade

**Comment by gardc** [comment]
Solved it by downgrading to python 3.8
